transit_odp/organisation/csv/service_codes.py

```diff
@@ -1,59 +1,229 @@
+                "line_number": service.service_number,
+                "dataset_id": file_attribute and file_attribute.revision.dataset_id,
+                "seasonal_status": seasonal_service
+                and seasonal_service.seasonal_status,
+                "seasonal_start": seasonal_service and seasonal_service.start,
+                "seasonal_end": seasonal_service and seasonal_service.end,
+                "staleness_status": staleness_status,
+                "effective_seasonal_start_date": seasonal_service
+                and seasonal_service.start - timedelta(days=42),
+                "effective_stale_date_end_date": file_attribute
+                and file_attribute.effective_stale_date_end_date,
+                "effective_stale_date_last_modified_date": file_attribute
+                and file_attribute.effective_stale_date_last_modified_date,
+                "last_modified_lt_effective_stale_date_otc": (
+                    service.last_modified.date()
+                    < service.effective_stale_date_otc_effective_date
+                ),
+                "effective_stale_date_otc_effective_date": (
+                    service.effective_stale_date_otc_effective_date
+                ),
+            }
         )
 
-    def get_queryset(self):
-        return OTCService.objects.get_all_without_exempted_ones(self._organisation_id)
+    def get_otc_service_data(self, organisation_id: int) -> None:
+        """
+        Compares an organisation's OTC Services dictionaries list with
+        TXCFileAttributes dictionaries list and the SeasonalService list
+        to determine which OTC Services require attention and which
+        doesn't ie. not live in BODS at all, or live but meeting new
+        Staleness conditions.
+        """
+        otc_map = get_otc_map(organisation_id)
+        txcfa_map = get_txc_map(organisation_id)
+        seasonal_service_map = get_seasonal_service_map(organisation_id)
 
-    def to_csv(self):
-        string_ = io.StringIO()
-        writer = csv.writer(string_, quoting=csv.QUOTE_ALL)
-        writer.writerow(
-            ("Service Status", "Licence Number", "Service Code", "Line", "Dataset ID")
-        )
+        for service_code, service in otc_map.items():
+            file_attribute = txcfa_map.get(service_code)
+            seasonal_service = seasonal_service_map.get(service_code)
+            if file_attribute is None:
+                require_attention = "Yes"
+                staleness_status = None
+                self._update_data(
+                    service,
+                    file_attribute,
+                    seasonal_service,
+                    staleness_status,
+                    require_attention,
+                )
+            elif requires_attention(service, file_attribute):
+                require_attention = "Yes"
+                rad = requires_attention_data(service, file_attribute)
+                staleness_status = STALENESS_STATUS[rad.index(True)]
```

Comment:
Are you sure that `rad` contains a `True` value? A service code can be not stale and still require attention, so it seems plausible. In this case `index` will raise a `ValueError`.

I think I would write this with a couple of `if` statements instead: more prosaic, but easier to follow.

Reply:
I am 100% never hit that exception because 
`rad = requires_attention_data(service, file_attribute)`
is behind an elif `elif requires_attention(service, file_attribute):`
if they are all `0` then this will be `False`